### PR TITLE
ls: only print colors if stdout is a tty

### LINKED ISSUE #1638: ls: don't use colors if stdout is not a TTY
Here is the relevant snippet in `configure`, at least on my Mac running Catalina

```
if (
   am_has_slept=no
   for am_try in 1 2; do
     echo "timestamp, slept: $am_has_slept" > conftest.file
     set X `ls -Lt "$srcdir/configure" conftest.file 2> /dev/null`
     if test "$*" = "X"; then
	# -L didn't work.
	set X `ls -t "$srcdir/configure" conftest.file`
     fi
     if test "$*" != "X $srcdir/configure conftest.file" \
	&& test "$*" != "X conftest.file $srcdir/configure"; then

	# If neither matched, then we have a broken ls.  This can happen
	# if, for instance, CONFIG_SHELL is bash and it inherits a
	# broken ls alias from the environment.  This has actually
	# happened.  Such a system could not be considered "sane".
	as_fn_error $? "ls -t appears to fail.  Make sure there is not a broken
  alias in your environment" "$LINENO" 5
     fi
     if test "$2" = conftest.file || test $am_try -eq 2; then
       break
     fi
     # Just in case.
     sleep 1
     am_has_slept=yes
   done
   test "$2" = conftest.file
   )
then
   # Ok.
```

I get the following error message when I attempt to compile this using coreutils `ls`

```
checking for a BSD-compatible install... /Users/kevin/local/coreutils/bin/install -c
checking whether build environment is sane... configure: error: ls -t appears to fail.  Make sure there is not a broken
  alias in your environment
configure: error: newly created file is older than distributed files!
Check your system clock
```

### PR DESCRIPTION
**Pull request by kevinburkemeter**
Previously if no --color argument was input, we would always print
colors in the output. This breaks `configure` scripts which run `ls`
and then compare the output against what they expect to see, since the
left side has ANSI escape sequences and the right side doesn't.

Instead, only print escape sequences if a TTY is present, or if
`--color=always` is specified.

Fixes #1638.

### CODE REVIEW

**Comment by sylvestre** [comment]
I guess writing a test for this is too hard?
Thanks

**Comment by sylvestre** [comment]
@kevinburkemeter  ping?

**Comment by kevinburkemeter** [comment]
Sorry. I am not a natural Rust developer, have probably written less than 1000 lines in my life. but am interested in this project because I'd prefer not to use glibc or C programs in general if I can help it. I'm not sure I'm going to have time to figure out how to add a test and ensure it works. I did verify locally that the patch behaves the way that I'd expect it to.

**Comment by sylvestre** [comment]
Please just give a try, it should be easy to verify in a test that there isn't any color commands :+1:

**Comment by sylvestre** [comment]
> I am not a natural Rust developer, have probably written less than 1000 lines in my life.

please note that, at some point, it has been the case of every one :)

File: src/uu/ls/src/ls.rs [hunk 1]
```diff
@@ -619,7 +619,7 @@ fn display_file_name(
     let mut width = UnicodeWidthStr::width(&*name);
 
     let color = match options.opt_str("color") {
-        None => true,
+        None => stdout_isatty(),
```
Comment (sylvestre) [inline src/uu/ls/src/ls.rs:622]: isatty is deprecated, I moved ls to atty

```suggestion
        None => atty::is(atty::Stream::Stdout),
```

**Comment by sylvestre** [comment]
@kevinburkemeter are you planning to finish this work? Thanks!

**Comment by kevinburkemeter** [comment]
Thank you for merging this and I'm really sorry I didn't have time to follow up.

**Pull request merged**